app/src/main/java/org/mozilla/fenix/library/bookmarks/BookmarkController.kt

```diff
@@ -158,6 +165,57 @@ class DefaultBookmarkController(
         showTabTray()
     }
 
+    private fun extractURLsFromTree(node: BookmarkNode): MutableList<String> {
+        val urls = mutableListOf<String>()
+
+        when (node.type) {
+            BookmarkNodeType.FOLDER -> {
+                // if not leaf (empty) folder
+                node.children?.forEach {
+                    urls.addAll(extractURLsFromTree(it))
+                }
+            }
+            BookmarkNodeType.ITEM -> {
+                node.url?.let { urls.add(it) }
+            }
+            BookmarkNodeType.SEPARATOR -> {}
+        }
+
+        return urls
+    }
+
+    override fun handleOpeningFolderBookmarks(folder: BookmarkNode, mode: BrowsingMode) {
+        // potentially heavy function with a lot of bookmarks to open => use a coroutine
+        scope.launch {
+            // if more than maximum elements
```

Comment:
I find myself thinking that a lot of these comments in the PR could be improved upon or some might not be necessary since they're just repeating what the code will tell you.

This comment unfortunately doesn't make sense in relation with the next line of code which is returning early when the bookmark folder cannot be loaded. Rather I think this comment was trying to say that we warn users if the number of bookmarks are bigger than `WARN_OPEN_ALL_SIZE`, but that is already stated further down.

Reply:
Agreed.  I'll remove most of the comments and leave the one that says "Warn user if more than maximum number of bookmarks are being opened"